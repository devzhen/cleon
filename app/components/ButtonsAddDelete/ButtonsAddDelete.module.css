.container {
  --margin-top: 0;
  --margin-right: 0;

  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin-top: var(--margin-top);
  margin-right: var(--margin-right);
}

.container button {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 57px;
  height: 31px;
  background-color: var(--color-main-background);
  cursor: pointer;
  color: #ffffff;
  user-select: none;
  font-size: 30px;
  font-weight: normal;
  line-height: 20px;
  text-transform: uppercase;
  border: none;
  outline: none;
}

.container button:active {
  box-shadow: 2px 2px 5px var(--color-main-background);
}

.container button:first-child {
  margin-bottom: 10px;
}
