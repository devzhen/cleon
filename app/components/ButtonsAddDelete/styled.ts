import styled from "styled-components";

export const Container = styled.div<{
  $marginTop?: number;
  $marginRight?: number;
}>`
  display: flex;
  flex-direction: column;
  align-items: flex-end;

  ${(props) => props.$marginTop && `margin-top: ${props.$marginTop}px`};
  ${(props) => props.$marginRight && `margin-right: ${props.$marginRight}px`};
`;

export const Button = styled.button<{
  $marginTop?: number;
}>`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 57px;
  height: 31px;
  background-color: ${(props) => props.theme.colors.mainBackground};
  cursor: pointer;
  color: ${(props) => props.theme.colors.white};
  user-select: none;
  font-size: 30px;
  font-weight: 400;
  line-height: 20px;
  text-transform: uppercase;
  border: none;
  outline: none;

  &:active {
    box-shadow: 2px 2px 5px ${(props) => props.theme.colors.mainBackground};
  }

  ${(props) => props.$marginTop && `margin-top: ${props.$marginTop}px`};
`;
