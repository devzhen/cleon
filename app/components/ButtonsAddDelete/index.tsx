import { Button, Container } from "./styled";

type ButtonAddDeleteProps = {
  add: () => void;
  remove: () => void;
  marginTop?: number;
  marginRight?: number;
};

const ButtonAddDelete = (props: ButtonAddDeleteProps) => {
  const { add, remove, marginRight, marginTop } = props;

  return (
    <Container $marginRight={marginRight} $marginTop={marginTop}>
      <Button onClick={add}>+</Button>
      <Button onClick={remove} $marginTop={10}>
        -
      </Button>
    </Container>
  );
};

export default ButtonAddDelete;
