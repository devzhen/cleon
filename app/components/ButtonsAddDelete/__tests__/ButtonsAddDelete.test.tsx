import { render } from '@testing-library/react';

import ButtonsAddDelete from '../';

describe('ButtonsAddDelete tests', () => {
  const props = {
    add: jest.fn(),
    remove: jest.fn(),
  };

  it('matches the snapshot', () => {
    const { container } = render(<ButtonsAddDelete {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot with margins', () => {
    const newProps = {
      ...props,
      marginTop: 100,
      marginRight: 100,
    };

    const { container } = render(<ButtonsAddDelete {...newProps} />);

    expect(container).toMatchSnapshot();
  });
});
