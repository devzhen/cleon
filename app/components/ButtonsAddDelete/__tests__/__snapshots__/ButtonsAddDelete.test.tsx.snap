// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsAddDelete tests matches the snapshot 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 0px; margin-right: 0px;"
  >
    <button>
      +
    </button>
    <button>
      -
    </button>
  </div>
</div>
`;

exports[`ButtonsAddDelete tests matches the snapshot with margins 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 100px; margin-right: 100px;"
  >
    <button>
      +
    </button>
    <button>
      -
    </button>
  </div>
</div>
`;
